internal/testutils/rust.go

```diff
@@ -74,35 +75,42 @@ func BuildRustNSSLib(t *testing.T, disableCoverage bool, features ...string) (li
 	cargo, isNightly, err := getCargoPath()
 	require.NoError(t, err, "Setup: looking for cargo")
 
-	// Note that for developing purposes and avoiding keeping building the rust program dependencies,
-	// TEST_RUST_TARGET environment variable can be set to an absolute path to keep iterative
-	// build artifacts.
-	target := os.Getenv("TEST_RUST_TARGET")
-	if target == "" {
-		target = t.TempDir()
+	// Store the build artifacts in a common temp directory, so that they can be reused between tests.
+	target := filepath.Join(os.TempDir(), "authd-tests-rust-build-artifacts")
+	if v := os.Getenv("TEST_RUST_TARGET"); v != "" {
+		target = v
 	}
 
+	err = os.MkdirAll(target, 0700)
+	require.NoError(t, err, "Setup: could not create Rust target dir")
+
 	rustDir := filepath.Join(projectRoot, "nss")
 	if !disableCoverage {
 		rustCovEnv = trackRustCoverage(t, target, rustDir)
 	}
 
 	features = append([]string{"integration_tests", "custom_socket"}, features...)
 
+	unlock, err := fileutils.LockDir(target)
+	require.NoError(t, err, "Setup: could not lock Rust target dir")
+	defer func() {
```

Comment:
I like `t.Cleanup` better for these things.

Reply:
Here we want to unlock the build directory when we leave `BuildRustNSSLib`, not when the test is done, because the test could take a long time, and we don't want to block other tests running in parallel.